modules/service-networking/main.tf

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_compute_global_address" "global_addresses" {
+  for_each      = { for address in var.global_addresses : address.name => address }
+  project       = var.project_id
+  name          = each.value.name
+  purpose       = each.value.purpose
```

Comment:
IIUC for this private service access usecase isn't it always `VPC_PEERING`?

Reply:
I have used VPC_PEERING as default value. Should i remove the variable and use VPC_PEERING directly?

Reply:
@q2w In case of service networking it will always be vpc_peering.